website/src/controllers/api-v3/groups.js

```diff
@@ -391,145 +392,101 @@ api.removeGroupMember = {
-        });
-      }
-    });
-  }))
-  .then(() => {
-    if (usersAlreadyRegistered.length > 0){
-      return _inviteByUUIDs(usersAlreadyRegistered, group, inviter, req, res, next);
+  if (group.type === 'guild') {
+    if (_.contains(userToInvite.guilds, group._id)) {
+      throw new NotAuthorized(res.t('userAlreadyInGroup'));
     }
+    if (_.find(userToInvite.invitations.guilds, {id: group._id})) {
+      throw new NotAuthorized(res.t('userAlreadyInvitedToGroup'));
+    }
+    userToInvite.invitations.guilds.push({id: group._id, name: group.name, inviter: res.locals.user._id});
+  } else if (group.type === 'party') {
+    if (!_.isEmpty(userToInvite.invitations.party)) {
+      throw new NotAuthorized(res.t('userAlreadyPendingInvitation'));
+    }
+    if (userToInvite.party._id) {
+      throw new NotAuthorized(res.t('userAlreadyInAParty'));
+    }
+    // @TODO: Why was this here?
+    // req.body.type in 'guild', 'party'
+    userToInvite.invitations.party = {id: group._id, name: group.name, inviter: res.locals.user._id};
+  }
 
-    res.respond(200, {}); // TODO what to return?
-  });
-}; */
+  let groupLabel = group.type === 'guild' ? 'Guild' : 'Party';
+  if (userToInvite.preferences.emailNotifications[`invited${groupLabel}`] !== false) {
+    let emailVars = [
+      {name: 'INVITER', content: inviter.name},
+      {name: 'REPLY_TO_ADDRESS', content: inviter.email},
+    ];
+
+    if (group.type === 'guild') {
+      emailVars.push(
+        {name: 'GUILD_NAME', content: group.name},
+        {name: 'GUILD_URL', content: '/#/options/groups/guilds/public'},
+      );
+    } else {
+      emailVars.push(
+        {name: 'PARTY_NAME', content: group.name},
+        {name: 'PARTY_URL', content: '/#/options/groups/party'},
+      );
+    }
+
+    sendTxnEmail(userToInvite, `invited-${groupLabel}`, emailVars);
+  }
+
+  let userInvited = await userToInvite.save();
+  return userInvited;
+}
+
+async function _inviteByEmail (invite, group, inviter, req, res) {
+  let userReturnInfo;
+
+  if (!invite.email) throw new BadRequest(res.t('inviteMissingEmail'));
+
+  let userToContact = await User.findOne({$or: [
+    {'auth.local.email': invite.email},
+    {'auth.facebook.emails.value': invite.email},
+  ]})
+  .select({_id: true, 'preferences.emailNotifications': true})
+  .exec();
+
+  if (userToContact) {
+    userReturnInfo = await _inviteByUUID(userToContact._id, group, inviter, req, res);
+  } else {
+    userReturnInfo = invite.email;
+    // yeah, it supports guild too but for backward compatibility we'll use partyInvite as query
+    // TODO absolutely refactor this horrible code
+    const partyQueryString = JSON.stringify({id: group._id, inviter, name: group.name});
+    const encryptedPartyqueryString = encrypt(partyQueryString);
+    let link = `?partyInvite=${encryptedPartyqueryString}`;
+
+    let variables = [
+      {name: 'LINK', content: link},
+      {name: 'INVITER', content: inviter || inviter.name},
```

Comment:
This doesn't look quite right to me. User is getting passed in here, so it should be `inviter.profile.name`